### PR TITLE
Cloudflare can use videodelivery.net subdomains

### PR DESCRIPTION
**Pull request by davex25**
Another resubmission of https://github.com/ytdl-org/youtube-dl/pull/21045, without code duplication.

### CODE REVIEW

**Comment by dstftw** [comment]
FYI, you should not flood with PRs but push into the same branch.

File: youtube_dl/extractor/cloudflarestream.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ class CloudflareStreamIE(InfoExtractor):
                     https?://
                         (?:
                             (?:watch\.)?cloudflarestream\.com/|
```
Comment (dstftw) [inline youtube_dl/extractor/cloudflarestream.py]: watch.videodelivery.net should also be added.

Comment (davex25) [reply youtube_dl/extractor/cloudflarestream.py]: Ok.

File: youtube_dl/extractor/cloudflarestream.py [hunk 2]
```diff
@@ -28,17 +28,26 @@ class CloudflareStreamIE(InfoExtractor):
     }, {
         'url': 'https://watch.cloudflarestream.com/9df17203414fd1db3e3ed74abbe936c1',
         'only_matching': True,
+    }, {
+        'url': 'https://watch.videodelivery.net/9df17203414fd1db3e3ed74abbe936c1',
+        'only_matching': True,
+    }, {
+        'url': 'https://videodelivery.net/9df17203414fd1db3e3ed74abbe936c1',
```
Comment (dstftw) [inline youtube_dl/extractor/cloudflarestream.py]: No such URL exists.

Comment (davex25) [reply youtube_dl/extractor/cloudflarestream.py]: True.  I don't have a working watch URL to add, should I just remove these?

Comment (dstftw) [reply youtube_dl/extractor/cloudflarestream.py]: Yes.

Comment (davex25) [reply youtube_dl/extractor/cloudflarestream.py]: Ok.

**Pull request merged**